packages/edit-site/src/components/page-patterns/style.scss

```diff
@@ -283,6 +283,10 @@
 		text-overflow: ellipsis;
 		color: inherit;
 	}
+
+	.dataviews-pagination {
+		z-index: 2;
```

Comment:
While this solves the problem, I couldn't identify the different stacking context that could cause the issue. So this feels hacky to me and would appreciate any suggestions, if there is a better fix.